src/test/java/com/google/api/codegen/testdata/java_test_test_library.baseline

```diff
@@ -0,0 +1,327 @@
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void listBooksTest() {
+    String formattedName = LibraryServiceApi.formatShelfName("[SHELF]");
+    PageAccessor<Book> response = api.ListBooks(formattedName);
+
+    // PageAccessor will not make actual request until it is being used.
+    // Add all the pages here in order to make grpc requests.
+    List<Book> resources = new ArrayList<>();
+    resources.addAll(response.getPageValues());
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    ListBooksRequest request = (ListBooksRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void deleteBookTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    api.DeleteBook(formattedName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    DeleteBookRequest request = (DeleteBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void updateBookTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    Book book = Book.newBuilder().build();
+    api.UpdateBook(formattedName, book);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    UpdateBookRequest request = (UpdateBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getBook(), book);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void moveBookTest() {
+    String formattedName = LibraryServiceApi.formatBookName("[SHELF]", "[BOOK]");
+    String formattedOtherShelfName = LibraryServiceApi.formatShelfName("[SHELF]");
+    api.MoveBook(formattedName, formattedOtherShelfName);
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    MoveBookRequest request = (MoveBookRequest)requests[0];
+
+    Assert.assertEquals(request.getName(), formattedName);
+    Assert.assertEquals(request.getOtherShelfName(), formattedOtherShelfName);
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void listStringsTest() {
+
+    PageAccessor<String> response = api.ListStrings();
+
+    // PageAccessor will not make actual request until it is being used.
+    // Add all the pages here in order to make grpc requests.
+    List<String> resources = new ArrayList<>();
+    resources.addAll(response.getPageValues());
+
+    List<GeneratedMessage> requests = serviceHelper.getService().getRequests();
+    Assert.assertEquals(1, requests.size());
+    ListStringsRequest request = (ListStringsRequest)requests[0];
+
+  }
```

Comment:
You don't have any support here for checking responses. The tests should be able to inject a response, and have the API return it back so that it can be checked.